examples/moses-pln-synergy/scm/README.md

```diff
@@ -76,16 +76,204 @@ And search for the following
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (EvaluationLink (stv 1 0.99999982)
+            (PredicateNode "contain")
+            (ListLink
+               (ConceptNode "treatment-1")
+               (ConceptNode "compound-A")
+            )
+         )
+      )
+   )
+...
+scheme@(guile-user)> (cog-bind lambda-grounded-construction-rule)
+$5 = (SetLink
+   (LambdaLink (stv 1 0.99999982)
+      (TypedVariableLink
+         (VariableNode "$X")
+         (TypeNode "ConceptNode")
+      )
+      (EvaluationLink (stv 1 0.99999982)
+         (PredicateNode "contain")
+         (ListLink
+            (ConceptNode "treatment-1")
+            (ConceptNode "compound-A")
+         )
+      )
    )
 )
 
+scheme@(guile-user)> (cog-bind implication-construction-rule)
+...
+   (ImplicationLink (stv 1 0.99999982)
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (EvaluationLink
+            (PredicateNode "take")
+            (ListLink
+               (VariableNode "$X")
+               (ConceptNode "treatment-1")
+            )
+         )
+      )
+      (LambdaLink (stv 1 0.99999982)
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (EvaluationLink (stv 1 0.99999982)
+            (PredicateNode "contain")
+            (ListLink
+               (ConceptNode "treatment-1")
+               (ConceptNode "compound-A")
+            )
+         )
+      )
+   )
+...
+
+scheme@(guile-user)> (cog-bind implication-implicant-distribution-rule)
+...
+   (ImplicationLink (stv 1 0.99999982)
+      (LambdaLink
+         (TypedVariableLink
+            (VariableNode "$X")
+            (TypeNode "ConceptNode")
+         )
+         (EvaluationLink
+            (PredicateNode "take")
+            (ListLink
+               (VariableNode "$X")
+               (ConceptNode "treatment-1")
+            )
+         )
+      )
+      (AndLink
+         (LambdaLink
+            (TypedVariableLink
+               (VariableNode "$X")
+               (TypeNode "ConceptNode")
+            )
+            (EvaluationLink
+               (PredicateNode "take")
+               (ListLink
+                  (VariableNode "$X")
+                  (ConceptNode "treatment-1")
+               )
+            )
+         )
+         (LambdaLink (stv 1 0.99999982)
+            (TypedVariableLink
+               (VariableNode "$X")
+               (TypeNode "ConceptNode")
+            )
+            (EvaluationLink (stv 1 0.99999982)
+               (PredicateNode "contain")
+               (ListLink
+                  (ConceptNode "treatment-1")
+                  (ConceptNode "compound-A")
+               )
+            )
+         )
+      )
+   )
+...
+scheme@(guile-user)> ;; If X takes treatment-1 then X takes compound-A
+scheme@(guile-user)> (cog-bind deduction-implication-rule)
```

Comment:
Just to be sure, note that the scopes of X do not overlap. Then, again PLN is not predicate logic, and Implication is not universally quantified, it is universally conditionally quantified, and to express this the Implication operates directly on predicates.